src/core/RandomVariables.jl

```diff
@@ -532,23 +532,26 @@ end
 end
 
 # Get the index (in vals) ranges of all the vns of variables belonging to spl
-function _getranges(vi::AbstractVarInfo, spl::Sampler)
-    if ~haskey(spl.info, :cache_updated) spl.info[:cache_updated] = CACHERESET end
-    if haskey(spl.info, :ranges) && (spl.info[:cache_updated] & CACHERANGES) > 0
-        spl.info[:ranges]
-    else
-        spl.info[:cache_updated] = spl.info[:cache_updated] | CACHERANGES
-        spl.info[:ranges] = _getranges(vi, spl.selector, spl.alg.space)
-    end
+@inline function _getranges(vi::AbstractVarInfo, spl::Sampler)
+    ## Uncomment the spl.info stuff when it is concretely typed, not Dict{Symbol, Any}
+    #if ~haskey(spl.info, :cache_updated) spl.info[:cache_updated] = CACHERESET end
+    #if haskey(spl.info, :ranges) && (spl.info[:cache_updated] & CACHERANGES) > 0
+    #    spl.info[:ranges]
+    #else
+        #spl.info[:cache_updated] = spl.info[:cache_updated] | CACHERANGES
+        ranges = _getranges(vi, spl.selector, getspace(spl.alg))
+        #spl.info[:ranges] = ranges
+        return ranges
+    #end
 end
 # Get the index (in vals) ranges of all the vns of variables belonging to selector `s` in `space`
-function _getranges(vi::AbstractVarInfo, s::Selector, space::Set=Set())
+@inline function _getranges(vi::AbstractVarInfo, s::Selector, space::Tuple=())
     __getranges(vi, _getidcs(vi, s, space))
 end
-function __getranges(vi::UntypedVarInfo, idcs)
+@inline function __getranges(vi::UntypedVarInfo, idcs)
     union(map(i -> vi.ranges[i], idcs)...)
 end
-__getranges(vi::TypedVarInfo, idcs) = __getranges(vi.metadata, idcs)
+@inline __getranges(vi::TypedVarInfo, idcs) = __getranges(vi.metadata, idcs)
```

Comment:
The `_f` pattern is definitely common. Here are a few cases of its usage in the Julia ecosystem:
1. Julia Base: https://github.com/JuliaLang/julia/blob/95fcde9dda4261ebd9dd2eb694b46fc74e472ef6/base/tuple.jl#L31
2. Tim Holy: https://github.com/timholy/ArrayIteration.jl/blob/master/src/core.jl#L110
3. Andy Ferris: https://github.com/JuliaArrays/StaticArrays.jl/blob/master/src/det.jl#L1
4. Kristoffer and Fredrik: https://github.com/KristofferC/JuAFEM.jl/blob/master/src/Dofs/DofHandler.jl#L273
5. Patrick and Asbjorn: https://github.com/JuliaNLSolvers/NLSolversBase.jl/blob/master/src/objective_types/constraints.jl#L192

The `__f` is just me taking that one step further. I am not sure if I have seen it before in other places or not. The idea of `_f` or `__f` is to take away the burden of reasoning about dispatch of `f` when reading the `_f` or `__f` functions or at least it has its own little dispatch rules. `_f` is supposed to be a more "internal" function than `f`, typically local to this one place in the files, and not to be used elsewhere hence no need to "waste" a new nice name on this function. 

Notice for example that `_getranges` and `__getranges` both have a 2 argument method. Trying to guess which method is called can be tricky without knowing the code inside out in order to know the types of the arguments and which method it will dispatch to. A trivially different name can therefore increase readability for foreigners reading the code. That's just my preference though. I can see the confusion if one is not familiar with this pattern. Perhaps, I can merge either `getranges` and `_getranges` or the other 2. Would you prefer that?

Reply:
Sure.